dask_ml/model_selection/_incremental.py

```diff
@@ -0,0 +1,288 @@
+        return model, {'ident': ident, 'params': params, 'time_step': -1}
+
+
+def inverse(start, batch):
+    """ Decrease target number of models inversely with time
+
+    This means that we train many models to start for a brief time and a few
+    models at the end for a long time
+    """
+    return int(start / (1 + batch))
+
+
+@gen.coroutine
+def _fit(
+    model,
+    params,
+    X_train,
+    y_train,
+    X_test,
+    y_test,
+    start=1000,
+    fit_params=None,
+    random_state=None,
+    scorer=None,
+    target=inverse,
+):
+    original_model = model
+    fit_params = fit_params or {}
+    client = default_client()
+    rng = check_random_state(random_state)
+    param_iterator = iter(ParameterSampler(params, start, random_state=rng))
+    target = functools.partial(target, start)
+
+    info = {}
+    models = {}
+    scores = {}
+
+    for ident in range(start):
+        params = next(param_iterator)
+        model = client.submit(_create_model, original_model, ident,
+                              random_state=rng.randint(2**31), **params)
+        info[ident] = {'params': params, 'param_index': ident}
+        models[ident] = model
+
+    # assume everything in fit_params is small and make it concrete
+    fit_params = yield client.compute(fit_params)
+
+    # Convert testing data into a single element on the cluster
+    # This assumes that it fits into memory on a single worker
+    if isinstance(X_test, da.Array):
+        X_test = client.compute(X_test)
+    else:
+        y_test = yield client.scatter(y_test)
+    if isinstance(y_test, da.Array):
+        y_test = client.compute(y_test)
+    else:
+        y_test = yield client.scatter(y_test)
+
+    # Convert to batches of delayed objects of numpy arrays
+    X_train = X_train.to_delayed()
+    if hasattr(X_train, 'squeeze'):
+        X_train = X_train.squeeze()
+    y_train = y_train.to_delayed()
+    if hasattr(y_train, 'squeeze'):
+        y_train = y_train.squeeze()
+    X_train, y_train = dask.optimize(X_train.tolist(), y_train.tolist())
+
+    # Create order by which we process batches
+    # TODO: create a non-repetitive random and uniform ordering
+    order = list(range(len(X_train)))
+    rng.shuffle(order)
+    seen = {}
+    tokens = {}
+
+    def get_futures(time_step):
+        j = order[time_step % len(order)]
+
+        if time_step < len(order) and j not in seen:  # new future, need to tell scheduler about it
+            X_future = client.compute(X_train[j])
+            y_future = client.compute(y_train[j])
+            seen[j] = (X_future.key, y_future.key)
+
+            # This is a hack to keep the futures in the scheduler but not in memory
+            X_token = client.submit(len, X_future)
+            y_token = client.submit(len, y_future)
+            tokens[time_step] = (X_token, y_token)
+
+            return X_future, y_future
+
+        else:
+            x_key, y_key = seen[j]
+            return Future(x_key), Future(y_key)
```

Comment:
I don't recommend looking too deeply at this function.  It's unclean and likely to go in the future.